### ISSUE TITLE
Not working with iOS 12.1

### ISSUE THREAD

**Issue by FD-** [dsafa22/AirplayServer#2]
I'm trying to use this app with iOS 12.1, but I could not get it to work. Whenever I try to connect to the AirPlay receiver of this app (called "t3" in my case), the iOS device shows the error `Could not connect to "t3"` (or similar, English is not the device language).

The iOS Console contains the following information:

```
default	13:34:56.886424 +0200	mediaserverd	<<<< FigRoutingManager_iOS >>>> FigRoutingManagerPickRouteDescriptorForContext: CMSession: Context type 'System Mirroring' picking route: t3
default	13:34:56.887311 +0200	mediaserverd	<<<< FigRoutingManager_iOS >>>> routingManager_preprocessPickEndpoint: CMSession: preprocessPickEndpoint - endpointName=t3, routingContextType=System Mirroring, clientPID=50, clientName=SpringBoard
default	13:34:56.887638 +0200	mediaserverd	<<<< FigRoutingManagerContextUtilities >>>> FigRoutingManagerContextUtilities_AddCurrentlyActivatingEndpoint: CMSession: context utilities: Add currently activating endpoint for endpointID=B4:CE:F6:07:17:8F, endpointName=t3, contextType=System Mirroring
default	13:34:56.888297 +0200	mediaserverd	<<<< FigRoutingManager_iOSEndpointHelpers >>>> FigRoutingManager_iOSActivateEndpoint: CMSession: Activating endpoint 't3' with features 7
default	13:34:56.888492 +0200	mediaserverd	[0xA9EC] Setting endpoint routing delegate 0x5CAB.
default	13:34:56.888659 +0200	mediaserverd	[0xA9EC] Setting endpoint remote control delegate 0x52AC.
default	13:34:56.888831 +0200	mediaserverd	<<<< FigRoutingManager >>>> FigRoutingManagerEndpointActivateWithCompletionCallback: CMSession: routingManager: endpointName = t3, inFeatures=7, inActivateOptions={
    IsPerAppDisplay = 0;
    PrimaryUsage = "PrimaryUsage_Screen";
    RoutingContextUUID = "153C071C-9D4E-4E71-BBC5-DE230A9C90CD";
}
default	13:34:56.890841 +0200	mediaserverd	Activating endpoint [0xA9EC] with features 0x7 and options {
    "IsPerAppDisplay" : false,
    "PrimaryUsage" : "PrimaryUsage_Screen",
    "RoutingContextUUID" : "153C071C-9D4E-4E71-BBC5-DE230A9C90CD",
}
default	13:34:56.894537 +0200	SpringBoard	[AVOutputContext] Received output context change initiated notification: NSConcreteNotification 0x281098ea0 {name = AVOutputContextDestinationChangeInitiatedNotification; object = <AVOutputContext: 0x281f318f0>; userInfo = {
    AVOutputContextDestinationChangeKey = "<AVOutputContextDestinationChange: 0x281f7c710, status=1, cancellationReason=(null)>";
}} for context: <MRAVOutputContext: 0x283e40730 type=SharedSystemScreen avOutputContext=<AVOutputContext: 0x281f318f0>>
default	13:34:56.895099 +0200	mediaserverd	PrimaryUsage use is not supported for use with PTP
default	13:34:56.895211 +0200	mediaserverd	receiverSupportsPTP 0
default	13:34:56.895495 +0200	mediaserverd	NTP use is not forced
default	13:34:56.895634 +0200	mediaserverd	Created APSNetworkClock NTP 0x555D
default	13:34:56.895930 +0200	mediaserverd	APSenderSessionAirPlay publishEncryptionKeyDiagnosticModeEnabled is off.
default	13:34:56.896219 +0200	mediaserverd	[0xA9EC] Activating endpoint 'AppleTV' (AppleTV2,1 v220.68) with seed 2...
default	13:34:56.914460 +0200	mediaserverd	-CMSMUtilities- CMSMUtility_PostNotificationToSession: CMSession: Notifications posted successfully. Overall system looks good.
default	13:34:56.921898 +0200	mediaremoted	[APSSystemUtils] BootUUID xxxxxxxx-xxxx-457C-9BC9-9A798ACF7EA5
default	13:34:56.926173 +0200	accessoryd	Pickable Routes Changed: <private>
default	13:34:56.926500 +0200	accessoryd	_pickableRoutesForPhoneCall is unsupported
default	13:34:56.926555 +0200	mediaremoted	[APSSystemUtils] BootUUID xxxxxxxx-xxxx-457C-9BC9-9A798ACF7EA5
default	13:34:56.927110 +0200	accessoryd	_pickableRoutesForVoiceMailWithForceNewRequest is unsupported
default	13:34:56.948218 +0200	mediaserverd	Unrecognized connectivity helper event 103.
default	13:34:56.948248 +0200	mediaserverd	Connect over Infra with "t3" ( senderSession->preferP2P = 1, isInfraAvailable = 1, isP2PAvailable = 0 )
default	13:34:56.951873 +0200	mediaserverd	APTransportConnectionHTTP 0xCECB with name t3-ctrl created. Connection QoS: 0
default	13:34:56.953272 +0200	wifid	WiFi:[577020896.953122]: Attached client mediaserverd is background application
default	13:34:56.954167 +0200	mediaserverd	APTransportConnectionHTTP 0xDE22 with name t3-rctl created. Connection QoS: 0
default	13:34:56.954293 +0200	wifid	WiFi:[577020896.953728]: Enable WoW requested by "mediaserverd"
default	13:34:56.956438 +0200	apsd	_WiFiManagerClientWoWStateChangedCallback WoW isEnabled YES WoW state <private>
default	13:34:56.957193 +0200	apsd	_WiFiManagerClientWoWStateChangedCallback WoW isEnabled YES WoW state <private>
default	13:34:56.958695 +0200	mediaserverd	[0xCECB] httpconnection_asyncConnectionProgressed: Connecting to address: [fe80::xxxx:xxxx:xxxx:178f%en0]:53601
default	13:34:56.958730 +0200	mediaserverd	APTransportTrafficRegistrar: MAC 00:00:00:00:00:00 has 0 connections, 1 require infrastructure
default	13:34:56.959066 +0200	kernel	com.apple.p2p: setAwdlPeerTrafficRegistration(): Clear traffic registration for airplay peer(00:00:00:00:00:00) hint=2
default	13:34:56.959091 +0200	kernel	com.apple.p2p: setAwdlPeerTrafficRegistration Airplaying on infra (6) has started
default	13:34:56.959129 +0200	kernel	com.apple.p2p: AWDL ON: [infra(6) 96%], (6/0/6) [6 0 6 0 0 0 0 0 6 0 6 0 0 0 0 0] Idle
default	13:34:56.959429 +0200	mediaserverd	APTransportTrafficRegistrar: Register AirPlay traffic for AWDL at MAC 00:00:00:00:00:00 with target infra critical PeerIndication=2 err=0
default	13:34:56.961375 +0200	symptomsd	L2 Metrics on ifname en0: rssi: -69 (txFrames/txReTx/txFail) 0/0/0 -> (was/is) 0/0
default	13:34:56.963330 +0200	callservicesd	Pickable Routes Changed: <private>
default	13:34:56.963445 +0200	callservicesd	_pickableRoutesForPhoneCall is unsupported
default	13:34:56.963790 +0200	callservicesd	_pickableRoutesForVoiceMailWithForceNewRequest is unsupported
default	13:34:56.964870 +0200	mediaserverd	
Interface en0 Peer 00:00:00:00:00:00: RSSI value = 0 , 
Interface en0 Peer FF:FF:FF:FF:FF:FF: RSSI value = 0 , 
Interface en0: Expected Peak Latency = 0 ms, LQM RSSI value = -69 , Effective Data Transfer Rate = 26 Mi, Interface Open Percent = 100 , Interface Avgerage CCA = 5 , LQM SNR value = 24 , 
Interface awdl0: Expected Peak Latency = 0 ms, Interface Open Percent = 24 , Interface Avgerage CCA = 5 ,
default	13:34:56.969218 +0200	callservicesd	Pickable Routes Changed: <private>
default	13:34:57.078928 +0200	mediaserverd	[0xCECB] httpconnection_asyncConnectionProgressed: Connecting to address: 192.168.178.72:53601
default	13:34:57.128637 +0200	mediaserverd	[0xCECB] httpConnection_setupHTTPClient: HTTPClient connected to 192.168.178.72:53601 (from 192.168.178.49:51508)
error	13:34:57.161971 +0200	mediaserverd	VBAEFt:618: false condition
default	13:34:57.162059 +0200	mediaserverd	### Pair-verify failed: -6743/0xFFFFE5A9 kSizeErr
default	13:34:57.162125 +0200	mediaserverd	### Control pair-verify legacy failed: -6743/0xFFFFE5A9 kSizeErr
default	13:34:57.162219 +0200	mediaserverd	Pair-setup over Infra with "t3"
default	13:34:57.177791 +0200	securityd	SecDbKeychainItemV7: item decrypted succussfully, but has no secret data so it's useless
default	13:34:57.180044 +0200	securityd	SecDbKeychainItem: encrypting item with no secret data
default	13:34:57.184373 +0200	securityd	replaced <genp,rowid=692,cdat=2019-04-15 11:33:48 +0000,mdat=2019-04-15 11:34:20 +0000,desc=AirPlay Client Peer,icmt=null,crtr=null,type=1634747216,scrp=null,labl=AirPlay Client Peer: 842e6d7f,alis=null,invi=null,nega=null,cusi=null,prot=null,acct=842e6d7fc2261e03a8e554fc0047xxxxxxxxxxxxxxxxea3643d7ddc9be497262,svce=AirPlay Client Peer,gena=null,data=0819:070000000ACF040A...|4ac16ea487dbf6e4,agrp=com.apple.airplay,pdmn=dk,sync=0,tomb=0,sha1=2940A1CFB98735AC763FD6F6E305ED321DF8F168,vwht=null,tkid=null,v_Data=<?>,v_pk=65150BB12CDC2XXXXXXXXFAFAFC907403A36381F,accc=310C300A0C04XXXX6F740C02646B,u_Tomb=null,musr=,UUID=042EB8C9-F048-XXXX-A74D-2290F011E342,sysb=null,pcss=null,pcsk=null,pcsi=null,persistref=> in <SecDbConnection rw open>
default	13:34:57.184932 +0200	securityd	    with <genp,rowid=692,cdat=2019-04-15 11:33:48 +0000,mdat=2019-04-15 11:34:57 +0000,desc=AirPlay Client Peer,icmt=null,crtr=null,type=1634747216,scrp=null,labl=AirPlay Client Peer: 842e6d7f,alis=null,invi=null,nega=null,cusi=null,prot=null,acct=842e6d7fc2261e03a8e554fc0047xxxxxxxxxxxxxxxxea3643d7ddc9be497262,svce=AirPlay Client Peer,gena=null,data=0818:070000000ACF040A...|41b48780f87ac813,agrp=com.apple.airplay,pdmn=dk,sync=0,tomb=0,sha1=B3930DA6516AAE25EF674593C48EECCC45E32823,vwht=null,tkid=null,v_Data=<?>,v_pk=65150BB12CDC2XXXXXXXXFAFAFC907403A36381F,accc=310C300A0C04XXXX6F740C02646B,u_Tomb=null,musr=,UUID=042EB8C9-F048-XXXX-A74D-2290F011E342,sysb=null,pcss=null,pcsk=null,pcsi=null,persistref=> in <SecDbConnection rw open>
default	13:34:57.185862 +0200	securityd	in/P+oxBmJFiD+nxOwIp82NpVA: will-commit api 1 changes, txn=<SecDbConnection rw open>, 0x102a0c260
default	13:34:57.187565 +0200	mediaserverd	Control pair-setup UA legacy success
default	13:34:57.221441 +0200	mediaserverd	Control pair-verify legacy success
default	13:34:57.221734 +0200	mediaserverd	elided platform fast path for key: re6Zb+zwFKJNlkQTUeT+/w
default	13:34:57.233153 +0200	mediaserverd	NTP server started on port 49812
default	13:34:57.233559 +0200	mediaserverd	Started APSNetworkClock NTP 0x555D
error	13:34:57.242799 +0200	mediaserverd	### [0xCECB] httpconnection_httpClientInvalidated: invalidate with reason -6753/0xFFFFE59F kConnectionErr.
error	13:34:57.243036 +0200	mediaserverd	[0xBD6F] Transport Stream 0x8AE3 disconnected
error	13:34:57.243314 +0200	mediaserverd	### Received sender session failed notification.
error	13:34:57.243511 +0200	mediaserverd	### [0xB1E4] stream 0x8AE3 disconnected with reason -6753/0xFFFFE59F kConnectionErr.
default	13:34:57.243726 +0200	mediaserverd	[0xBD6F] apsession_handleTransportSessionDisconnected: transport session disconnected with reason: Unknown error: -16613. Session started = false
error	13:34:57.243862 +0200	mediaserverd	### Received sender session failed notification.
error	13:34:57.244189 +0200	mediaserverd	stream_SendMessageCreatingReply:1373: got error -16612/0xFFFFBF1C Unknown error: -16612
error	13:34:57.244374 +0200	mediaserverd	APTransportStreamSendPlistMessageCreatingPlistReply:67: got error -16612/0xFFFFBF1C Unknown error: -16612
error	13:34:57.244954 +0200	mediaserverd	apsession_fetchInfo:3425: got error -16612/0xFFFFBF1C Unknown error: -16612
error	13:34:57.245100 +0200	mediaserverd	apsession_ensureInfoFetched:3382: got error -16612/0xFFFFBF1C Unknown error: -16612
error	13:34:57.245459 +0200	mediaserverd	apsession_ensureStartedInternal:1885: got error -16612/0xFFFFBF1C Unknown error: -16612
default	13:34:57.245766 +0200	mediaserverd	Disconnecting from "t3"
default	13:34:57.246261 +0200	mediaserverd	NTP server stopped
default	13:34:57.246388 +0200	mediaserverd	Stopped APSNetworkClock NTP 0x555D
error	13:34:57.247420 +0200	mediaserverd	endpoint_activateInternal:2533: got error -16612/0xFFFFBF1C Unknown error: -16612
error	13:34:57.247546 +0200	mediaserverd	[0xA9EC] Endpoint activation (seed 2) failed with error -16612/0xFFFFBF1C Unknown error: -16612.
default	13:34:57.247863 +0200	mediaserverd	[0xA9EC] Deactivating endpoint 't3' with reason 'FailedActivation'...
```

**Comment by dsafa22** [comment]
Looks like a handshake error. Can you provide the logs of Android when it happens?

**Comment by FD-** [comment]
It seems the problem only occurs after tapping the "END" button and then "START" again. Initially, it works (while the device name is t1).

When it fails, this is the output on the Android side:
```
9-04-19 17:48:39.016 4998-4998/com.fang.myapplication D/DNSNotify: registerAirplay port = 51566, mMacAddress = B4:CE:F6:07:17:8F
2019-04-19 17:48:39.021 4998-4998/com.fang.myapplication D/raop: raop init success
2019-04-19 17:48:39.021 4998-4998/com.fang.myapplication I/raop: Initialized server socket(s)
2019-04-19 17:48:39.021 4998-4998/com.fang.myapplication D/raop: raop port =  54445
2019-04-19 17:48:39.021 4998-4998/com.fang.myapplication D/DNSNotify: registerRaop port = 54445
2019-04-19 17:48:39.022 4998-4998/com.fang.myapplication D/MainActivity: airplayPort = 51566, raopPort = 54445
2019-04-19 17:48:39.720 4998-5050/com.fang.myapplication I/TAG: Register successfully : t4
2019-04-19 17:48:39.720 4998-5052/com.fang.myapplication I/TAG: Register successfully : B4CEF607178F@t4
2019-04-19 17:48:44.176 4998-5051/com.fang.myapplication I/raop: Accepted IPv4 client on socket 67
2019-04-19 17:48:44.176 4998-5051/com.fang.myapplication I/raop: Local: 192.168.1.7
2019-04-19 17:48:44.176 4998-5051/com.fang.myapplication I/raop: Remote: 192.168.1.11
2019-04-19 17:48:44.177 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.177 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.177 4998-5051/com.fang.myapplication D/raop: Handling request GET with URL /info
2019-04-19 17:48:44.190 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.191 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.191 4998-5051/com.fang.myapplication D/raop: Handling request POST with URL /pair-setup
2019-04-19 17:48:44.218 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.218 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.218 4998-5051/com.fang.myapplication D/raop: Handling request POST with URL /pair-verify
2019-04-19 17:48:44.327 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.328 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.328 4998-5051/com.fang.myapplication D/raop: Handling request POST with URL /pair-verify
2019-04-19 17:48:44.335 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.336 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.336 4998-5051/com.fang.myapplication D/raop: Handling request POST with URL /fp-setup
2019-04-19 17:48:44.346 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.346 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.347 4998-5051/com.fang.myapplication D/raop: Handling request POST with URL /fp-setup
2019-04-19 17:48:44.357 4998-5051/com.fang.myapplication D/raop: Receiving on socket 67
2019-04-19 17:48:44.357 4998-5051/com.fang.myapplication D/raop: conn_request
2019-04-19 17:48:44.357 4998-5051/com.fang.myapplication D/raop: Handling request SETUP with URL rtsp://192.168.1.7/17949857292363779043
2019-04-19 17:48:44.358 4998-5051/com.fang.myapplication D/raop: DACP-ID: 2206580501B7EA18
2019-04-19 17:48:44.358 4998-5051/com.fang.myapplication D/raop: Active-Remote: 3200590909
2019-04-19 17:48:44.358 4998-5051/com.fang.myapplication I/raop: Transport: null
2019-04-19 17:48:44.358 4998-5051/com.fang.myapplication D/raop: SETUP 3
2019-04-19 17:48:44.358 4998-5051/com.fang.myapplication E/raop: RAOP not initialized at SETUP, playing will fail!
2019-04-19 17:48:44.359 4998-5051/com.fang.myapplication D/raop: SETUP 3 len = 113
2019-04-19 17:48:44.359 4998-5051/com.fang.myapplication I/raop: dport = 0, tport = 0, cport = 0
2019-04-19 17:48:44.360 4998-5051/com.fang.myapplication I/raop: Disconnecting on software request
```

**Comment by dsafa22** [comment]
I seem to know what's wrong and I'll fix it as soon as possible.

**Comment by dsafa22** [comment]
This problem has been fixed and verified by IOS 12.2.
